index.js

```diff
@@ -45,63 +52,43 @@ function disconnect() {
  * Connect to a mongodb
  *
  * @param {string} mongoConnectionString complete mongo connection string. e.g. user:password@mongodb:27017/database
- * @param {object} [options] mongo client options see link below
- * https://mongodb.github.io/node-mongodb-native/1.4/driver-articles/mongoclient.html?highlight=server#db-a-hash-of-options-at-the-db-level-overriding-or-adjusting-functionality-not-supported-by-the-url
+ * @param {object} [options] mongo client options; see README
  *
- * @returns {Boolean} If a new connection has been established
+ * @returns {Promise<Mongoose>} Returns the connection
  */
 function connect(mongoConnectionString, options = {}) {
   // Generate options
   options = { ...defaultOptions, ...options }
-  logger.info('**** mongoose-harakiri: Connecting with these options;', options)
+  logger.info('**** mongoose-harakiri: Connecting with these options:')
+  logger.info('**** mongoose-harakiri: ', options)
 
   // Check if connection is already open
   if (mongoose.connection.readyState > 0) {
     logger.debug('**** mongoose-harakiri: Connection is already open.')
-    return false
+    return Promise.resolve(mongoose.connection)
   }
 
   // Connect
   try {
-    mongoose.connect(mongoConnectionString, options)
+    return mongoose.connect(mongoConnectionString, options)
   } catch (error) {
     logger.error('**** mongoose-harakiri: Failed connecting to the DB!')
     logger.error(error)
     process.exit(1)
   }
-
-  // apply listeners
-  try {
-    applyListenersToConnection(mongoose.connection)
-  } catch (error) {
-    logger.error('**** mongoose-harakiri: Could not apply listeners!')
-    logger.error(error)
-    process.exit(1)
-  }
-  return true
 }
 
 function applyListenersToConnection(connection) {
   connection.on('disconnected', () => {
     logger.info('**** mongoose-harakiri: Default connection disconnected.')
+    if (killOnDisconnect) return process.exit(1)
+    killOnDisconnect = true // It should die once it would disconnect again
```

Comment:
why on second disconnect?

and why isn't this an option we take from outside? if someone doesn't use sharded clusters, it shouldn't die on disconnect but rather try to reconnect first.

Reply:
I changed it to be an option that can be passed in.